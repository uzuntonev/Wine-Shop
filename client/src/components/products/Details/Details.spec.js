import React from 'react';
import Details from './Details';
import renderer from 'react-test-renderer';
import { createMuiTheme, ThemeProvider } from '@material-ui/core/styles';
import Context from '../../../store/store';

import { BrowserRouter } from 'react-router-dom';
describe('Component "Details"', () => {
  let tree;
  let useContextSpy;
  let theme;
  let instance;
  beforeEach(() => {
    useContextSpy = jest
      .spyOn(React, 'useContext')
      .mockReturnValueOnce({ state: { products: [{},{}] } });
    theme = createMuiTheme({
      palette: {
        primary: {
          main: '#121212',
          text: '#B0976D',
        },
      },
    });
    const component = renderer.create(
      <ThemeProvider theme={theme}>
        <Context>
          <BrowserRouter>
            <Details />
          </BrowserRouter>
        </Context>
      </ThemeProvider>
    );
    let tree = component.toJSON();
    let instance = component.root;
  });

  it('component "Details" is mount', () => {
    expect(tree).toMatchSnapshot();
  });

  it('should called useContext', () => {
    expect(useContextSpy).toHaveBeenCalledTimes(16);
  });

//   it('Should called AddToCard after click', (done) => {
//       const component = renderer.create(

//               <Details />

//       );
//       let tree = component.toJSON();
//       let instance = component.root;
//       console.log(component.root.props);

//   });
});
