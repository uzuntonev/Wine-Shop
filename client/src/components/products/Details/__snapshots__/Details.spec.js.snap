// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component "Details" component "Details" is mount 1`] = `undefined`;
